wyoming_whisper_trt/handler.py

```
@@ -136,30 +158,72 @@ async def handle_event(self, event: Event) -> bool:
-                self._wave_writer.setsampwidth(chunk.width)
-                self._wave_writer.setnchannels(chunk.channels)
-                logger.debug("Initialized in-memory WAV buffer.")
-            except wave.Error as e:
-                logger.error("Failed to initialize in-memory WAV buffer: %s", e)
-                raise
-        # Write the chunk's audio data.
 
+        # 1) init WAV writer if needed
+        if self._wave_writer is None:
+            self._wav_buffer = io.BytesIO()
+            self._wave_writer = wave.open(self._wav_buffer, "wb")
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)